### ISSUE TITLE
Time outs when run from Amazon EC2 instances

### ISSUE THREAD

**Issue by jugglinmike** [w3c/wptdashboard#441]
Running the Web Platform Tests with Firefox and Chrome is possible on Amazon EC2 instances. It's slightly counter-intuitive that the test should time out for the Edge and Safari browsers because the browser processes themselves are running remotely (on servers provided by Sauce Labs). This may be related to network latency which is not present when running the tests against locally-installed browsers.

Workaround: we've deploy those workers on Bytemark servers

**Comment by mattl** [comment]
So I've done a bunch of reading on this, and I believe we could be succesful running things on AWS but we were using the wrong type of EC2 machines. T machines are burstable in terms of their performance, so ideal for a situation like running a web server where you don't have constant visitors. 

I'd like to try something: running the Firefox tests on both a t2 and m5 machine.

**Comment by mattl** [comment]
Launching at m5.large machine now.  i-0a12b0819c35efb69 
Also launching a t2.large machine now. i-038fe54e00473be55

Both machines have 2 CPU and 8GB RAM.

**Comment by mattl** [comment]
m5.large = 34.235.126.144
t2.large = 54.234.39.146

Just spent about 20 minutes setting up wptdashboard and web-platform-test repos, plus Firefox 58.0. The full test suite is running (from the fix-firefox-paths branch) inside GNU Screen (screen -D -RR to monitor it) -- on screen 0, the tests are running (with xvfb), while on screen 1 the following command is running ``watch -n 60 "vmstat >> ~/vmstats.txt && date >> ~/vmstats.txt && uptime >> ~/vmstats.txt"`` -- and on screen 2, we can see a tail of that file.

**Comment by mattl** [comment]
@jugglinmike i think i found a regression in run.py -- when you're in later today I'll show you.

**Comment by mattl** [comment]
The regression was here https://github.com/w3c/wptdashboard/pull/488/commits/c2b6f275bc318190003ad1d3e9a40493a88efaac -- it has been fixed and the machine is running. It should be finished in the morning.

**Comment by mattl** [comment]
The numbers I got back on the t2 machine aren't helpful as the wpt process stalled. Running it again.

**Comment by jugglinmike** [comment]
In the process of [distributing the test running process across many machines](https://github.com/w3c/wptdashboard/pull/522), we experimented with collecting results using Amazon EC2, and this time around, we found success.  It's not clear what was originally causing problems here, but [our research suggests that neither AWS "burst credits" nor network throughput are to blame](https://bocoup.github.io/presentation-wptd-parallelization).

I'm closing this because things now appear to be working as expected.
